### PR TITLE
Cap container hostname length to 63 chars.

### LINKED ISSUE #5829: Failed to start container: unable to sethostname when calling
The e2e tests started failing sometime this afternoon because docker was not able to set the hostname.

Sample e2e run: go/k8healthz/job/kubernetes-e2e-gce/3797/consoleFull
The log is full of sethostname errors:
```
Error syncing pod, skipping: API error (500): Cannot start container 48c57647f75fba98c3ed7d82574e3fdad936d87c4f352773bf7c5a1b7fe6d5c1: unable to sethostname "fluentd-to-elasticsearch-e2e-test-jenkins-minion-4gnu.c.kubernetes-jenkins.internal": invalid argument
```
kubelet uses `pod.name` as the container hostname (I am not sure why). In the example above, the hostname is 83 characters, which has exceeded the max length of 64 on the system.

The cause of this is that static pods (i.e., pods from file/http sources) use `podName-hostname` as the actual pod name so that we can distinguish pods from different nodes. #5775 replaced os.Hostname() with kubelet.Hostname. This led to the change of hostname from
`e2e-test-jenkins-minion-4gnu` to `e2e-test-jenkins-minion-4gnu.c.kubernetes-jenkins.internal`, thus exceeding the limit.

### PR DESCRIPTION
**Pull request by vmarmol**
Fixes #5829.

### CODE REVIEW

**Comment by vmarmol** [comment]
Verified failing e2e tests pass. PTAL @yujuhong

**Comment by brendandburns** [comment]
Validated that this fixes e2e on my client.  Merging out of hours since e2e is broken.

Thanks for the quick fix!

**Comment by yujuhong** [comment]
@vmarmol, thanks for fixing this!

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -765,13 +765,20 @@ func (kl *Kubelet) runContainer(pod *api.Pod, container *api.Container, podVolum
 	binds := makeBinds(container, podVolumes)
 	exposedPorts, portBindings := makePortsAndBindings(container)
 
+	// TODO(vmarmol): Handle better.
```
Comment (gserra-olx) [inline pkg/kubelet/kubelet.go]: coff coff... any follow up on this? :) @vmarmol ?

Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: Nope, but feel free to upload a PR if you have any ideas.

**Pull request merged**